src/tests/ftest/cat_recovery/ddb.py

```diff
@@ -0,0 +1,666 @@
+        # node. clush will append .<server_hostname> to the file when copying.
+        args = "--rcopy {} --dest {}".format(remote_file_path, self.test_dir)
+        clush_command = get_clush_command(hosts=self.hostlist_servers[0], args=args)
+        try:
+            run_command(command=clush_command)
+        except DaosTestError as error:
+            raise CommandFailure(
+                "ERROR: Copying {} from {}: {}".format(
+                    remote_file_path, self.hostlist_servers[0], error)) from error
+
+        # Remove the appended .<server_hostname> from the copied file.
+        current_file_path = "".join([remote_file_path, ".", self.hostlist_servers[0]])
+        mv_command = "mv {} {}".format(current_file_path, remote_file_path)
+        try:
+            run_command(command=mv_command)
+        except DaosTestError as error:
+            raise CommandFailure(
+                "ERROR: Moving {} to {}: {}".format(
+                    current_file_path, remote_file_path, error)) from error
+
+    def test_ls(self):
+        """Test ddb ls.
+
+        1. Verify container UUID.
+        2. Verify object count in the container.
+        3. Verify there are two dkeys for every object. Also verify the dkey string and
+        the size.
+        4. Verify there is one akey for every dkey. Also verify the key string and the
+        size.
+        5. Verify data length in every akey.
+        6. Restart the server for the cleanup.
+        7. Reset the container and the pool to prepare for the cleanup.
+
+        :avocado: tags=all,weekly_regression
+        :avocado: tags=vm
+        :avocado: tags=cat_rec
+        :avocado: tags=ddb,ddb_ls
+        """
+        # Create a pool and a container.
+        self.add_pool()
+        self.add_container(pool=self.pool)
+
+        # Find the vos file name. e.g., /mnt/daos/<pool_uuid>/vos-0.
+        ddb_command = DdbCommand(
+            server_host=NodeSet(self.hostlist_servers[0]), path=self.bin,
+            mount_point="/mnt/daos", pool_uuid=self.pool.uuid,
+            vos_file=self.get_vos_file_path())
+
+        errors = []
+
+        # Insert objects with API.
+        object_count = self.params.get("object_count", "/run/*")
+        dkey_count = self.params.get("dkey_count", "/run/*")
+        akey_count = self.params.get("akey_count", "/run/*")
+        self.insert_objects(
+            object_count=object_count, dkey_count=dkey_count, akey_count=akey_count)
+
+        # Need to stop the server to use ddb.
+        self.get_dmg_command().system_stop()
+
+        # 1. Verify container UUID.
+        cmd_result = ddb_command.list_component()
+        # Sample output.
+        # [0] d4e0c836-17bd-4df3-b255-929732486bab
+        # stdout is a list which contains each line as separate element. Concatenate them
+        # to single string so that we can apply regex.
+        ls_out = "\n".join(cmd_result[0]["stdout"])
+        match = re.findall(r"(\[\d+\])\s+([a-f0-9\-]+)", ls_out)
```

Comment:
You should be able to just do this
```suggestion
        match = re.findall(r"(\[\d+\])\s+([a-f0-9\-]+)", cmd_result[0]["stdout"], re.MULTILINE)
```

Reply:
`cmd_result[0]["stdout"]` is a list that contains each line as separate element, so this approach didn't work.